addon/mixins/sortable-item.js

```diff
@@ -714,5 +729,35 @@ export default Mixin.create({
     this.set('isDropping', false);
     this.set('wasDropped', true);
     this._tellGroup('commit');
-  }
+  },
+
+  /**
+   * Method used to move item 1 step up or down
+   *
+   * @method _move
+   * @param {Number|Direction} 1 for moving down (or right) -1 for opposite
+   * @private
+   */
+  _move(delta){
+    let groupDirection = this.get('group.direction');
+    let sortedItems = this.get('group.sortedItems');
+    let thisIndex = sortedItems.indexOf(this);
+
+    if (delta === -1) {
+      if (thisIndex === 0) {
+        return ;
+      }
+      let swapItem = sortedItems[thisIndex + delta];
+      this.set(groupDirection, swapItem.get(groupDirection) + delta);
+    } else {
+      if (thisIndex === (sortedItems.length - 1) ) {
+        return ;
+      }
+      let swapItem = sortedItems[thisIndex + delta];
+      this.set(groupDirection, swapItem.get(groupDirection) + delta);
+    }
```

Comment:
It seems a little indirect to me that we’re dealing in terms of coordinates to perform this operation, when really it’s an operation on an array. Could we somehow skip directly to producing the new array in the order and sending it out with the action?

Reply:
OK 🤔

Reply:
@H1D by the way — I’m very happy to be persuaded that this is the best approach. Questioning more than pronouncing :)

Reply:
OK, but I think you are right. This pixel arithmetics looks bad.

Do not see elegant solution here yet. Seems like `{{sortable-group}}` refactoring must be done for that